src/zenml/config/source.py

```diff
@@ -69,6 +69,11 @@ class Source(BaseModel):
     attribute: Optional[str] = None
     type: SourceType
 
+    dynamic_decorator_module: Optional[str] = None
+    dynamic_decorator_attribute: Optional[str] = None
+    dynamic_decorator_type: Optional[SourceType] = None
+    dynamic_decorator_kwargs: Optional[Dict[str, Any]] = {}
```

Comment:
Not a big fan of this part - it should have been a Source instance with Kwargs aside, but this leads to circular issues of Pydantic, while it tries to dump Source.

Reply:
Yeah agreed, I also don't really like that such a specific feature (including some config options for the decorator 😅) is now on this super basic class which is essentially just representing a string like "my_module.SomeClass".

It would be nice to handle this in a generic way, not just for decorators on steps, but have a way for objects to modify the behavior of source resolving of themselves/other objects.

Reply:
But even if we had it generic solution, I'm still not sure where we would store the config for this. For me, it seems like something that should be in the `PipelineDeployment` model. Maybe when the decorator is applied, it could add its configuration dynamically to the step configuration somehow?